apps/sports-helsinki/src/domain/search/combinedSearch/hooks.ts

```diff
@@ -0,0 +1,163 @@
+import { useLazyQuery } from '@apollo/client';
+import type {
+  SearchListQuery,
+  EventListQuery,
+} from 'events-helsinki-components';
+import {
+  SearchListDocument,
+  EventTypeId,
+  EventListDocument,
+} from 'events-helsinki-components';
+import { useRouter } from 'next/router';
+import React, { useRef, useEffect } from 'react';
+import AppConfig from 'domain/app/AppConfig';
+import useUnifiedSearchVariables from 'domain/unifiedSearch/useUnifiedSearchVariables';
+import useEventSearchFilters from '../eventSearch/hooks/useEventSearchFilters';
+import { PARAM_SEARCH_TYPE } from './constants';
+import type { SearchTabId } from './searchTabs/tabsContext';
+import { isSearchTabId, useTabsContext } from './searchTabs/tabsContext';
+
+/**
+ * Use an URL parameter to scroll to
+ * the previously used search result card on a page change.
+ */
+export const useScrollToSearchResultItem = () => {
+  const router = useRouter();
+  const scrollTo = router.query?.scrollTo;
+  const listRef = useRef<HTMLUListElement | null>(null);
+
+  useEffect(() => {
+    const listElement = listRef.current;
+
+    if (scrollTo) {
+      const listItemElement = listElement?.querySelector(
+        decodeURIComponent(scrollTo.toString())
+      );
+
+      if (listItemElement) {
+        listItemElement.scrollIntoView({
+          block: 'center',
+        });
+      }
+    }
+  }, [scrollTo]);
+};
+
+export const useSearchTabsWithParams = (defaultTab: SearchTabId) => {
+  const router = useRouter();
+  const searchParams = new URLSearchParams(router.asPath.split('?')[1]);
+  const searchTypeParam = searchParams.get(PARAM_SEARCH_TYPE);
+  const initTab = searchTypeParam
+    ? (searchTypeParam as SearchTabId)
+    : defaultTab;
+
+  // If the search type param is not given in the URL, set it there.
+  React.useEffect(() => {
+    if (!isSearchTabId(searchTypeParam)) {
+      router.push(
+        { query: { ...router.query, [PARAM_SEARCH_TYPE]: defaultTab } },
+        undefined,
+        { shallow: true }
+      );
+    }
+  }, [searchTypeParam, router, defaultTab]);
+
+  return { initTab, searchTypeParam };
+};
+
+export function useLazyVenueSearchForTabCount() {
```

Comment:
why some of these are regular functions and some arrow functions? It might confuse the reader a bit

Reply:
There should not be a file hooks.ts anymore, so now it does not confuse 😃 But anyway, I don't know what was the actual case. I just didn't think about that at all. I self prefer the basic functions.